### DISCUSSION TITLE
Using API to list characters available.

### DISCUSSION THREAD

**Discussion by Jewbby** [oobabooga/text-generation-webui#3159] [Q&A]
I'm trying to pull a list of the available characters to load. The round about way I've been testing with is just listing all characters available manually and updating the list as manual which gets passed once a selection is made. However I would like to be able to do this automatically if possible through a call. I did not see anything like this available in the examples. Anyone know of a way to achieve this if possible?

**Comment by boredofnames** [comment]
There doesn't seem to be an endpoint for it in the [code](https://github.com/oobabooga/text-generation-webui/blob/main/extensions/api/blocking_api.py). However, you could just rip the code out of the character [gallery](https://github.com/oobabooga/text-generation-webui/blob/9f0803886451c9b6e817c8b3adbe1044e6856d6e/extensions/gallery/script.py#L60C8-L60C8) setup, and modify it into an endpoint.

**Reply by Jewbby** [reply]
I'll take a look at this.

**Comment by matatonic** [comment]
I created this a while ago, but the PR has not been merged (been a while): https://github.com/oobabooga/text-generation-webui/pull/2618

**Reply by Jewbby** [reply]
I think I looked at that yesterday. It would be so nice.
